### PR TITLE
Add list command for shell script purpose, also add check in add habit to avoid duplicates on name

### PR DESCRIPTION
**Pull request by yoms**
Usage dijo -l for printing the habit names list

ps: Its my first commit in rust...

### CODE REVIEW

**Comment by NerdyPepper** [comment]
> Its my first commit in rust...

I am really happy to see people using `dijo`'s codebase as a place to start out learning!

File: src/app/impl_self.rs [hunk 1]
```diff
@@ -33,8 +33,27 @@ impl App {
         };
     }
 
+    pub fn list_habit(&self) -> Vec<String> {
+        let mut habits_names: Vec<String> = vec![];
```
Comment (NerdyPepper) [inline src/app/impl_self.rs:37]: I would suggest trimming this down to not require a mutable vector:
```suggestion
        let habits_names = self.habits.iter().map(|x| x.name()).collect::<Vec<_>>();
```

File: src/app/impl_self.rs [hunk 2]
```diff
@@ -33,8 +33,27 @@ impl App {
...
-        self.habits.push(h);
+        if self
+            .habits
+            .iter()
+            .filter(|hab| hab.name() == h.name())
+            .count()
+            > 0
+        {
+            self.message
+                .set_message(format!("Habit `{}` allready exist", h.name()))
```
Comment (NerdyPepper) [inline src/app/impl_self.rs:53]: Good catch! I completely forgot about this. Eventually, we can use some form of UUID to identify habits, but this is good for now!
I would suggest adding:
```rust
self.message.set_kind(MessageKind::Error);
```

File: src/main.rs [hunk 3]
```diff
@@ -33,6 +33,14 @@ fn main() {
                 .value_name("CMD")
                 .help("run a dijo command"),
         )
+        .arg(
```
Comment (NerdyPepper) [inline src/main.rs:36]: I would like to expand this to a full blown query system, using format strings.

Comment (yoms) [reply src/main.rs:36]: Do you need that for this PR or is it an idea for the future?

Comment (NerdyPepper) [reply src/main.rs:36]: This PR is good as it is, the query system and more command line flags can come later.

**Pull request merged**